tests/datasets/configuration/test_generic_metrics.py

```diff
@@ -1,6 +1,66 @@
+import pytest
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
 from yaml import safe_load
 
+from snuba.clickhouse.columns import AggregateFunction, Column, Float
+from snuba.datasets.configuration.json_schema import (
+    readable_storage_schema,
+    writable_storage_schema,
+)
+from snuba.datasets.configuration.utils import parse_columns
+from snuba.datasets.storages.generic_metrics import (
+    aggregate_common_columns,
+    bucket_columns,
+    common_columns,
+)
 
-def test_file_loads() -> None:
-    file = open("./snuba/datasets/configuration/generic_metrics/dataset.yaml")
-    safe_load(file)
+CONF_PATH = "./snuba/datasets/configuration/generic_metrics/storages"
+
+DISTRIBUTIONS_STORAGE_COLUMN_SET = [
+    *common_columns,
+    *aggregate_common_columns,
+    Column(
+        "percentiles",
+        AggregateFunction("quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]),
+    ),
+    Column("min", AggregateFunction("min", [Float(64)])),
+    Column("max", AggregateFunction("max", [Float(64)])),
+    Column("avg", AggregateFunction("avg", [Float(64)])),
+    Column("sum", AggregateFunction("sum", [Float(64)])),
+    Column("count", AggregateFunction("count", [Float(64)])),
+    Column(
+        "histogram_buckets",
+        AggregateFunction("histogram(250)", [Float(64)]),
+    ),
+]
+
+
+def test_distributions_storage() -> None:
+    file = open(f"{CONF_PATH}/distributions_storage.yaml")
+    config = safe_load(file)
+    validate(config, readable_storage_schema)
+    assert (
+        parse_columns(config["schema"]["columns"]) == DISTRIBUTIONS_STORAGE_COLUMN_SET
```

Comment:
If you are going to start this PR with stream loaders etc., you need to validate the entire storage not just columns.

Reply:
I've added a test for the stream loader, not sure if this is what you meant. Would we actually keep tests like these around? Seems kind of like rewriting the config but in test cases.

Reply:
Sorry for the confusion, I think there are a few things going on here.

1) You need to write tests for all the functionality you're adding. You have tests for column parsing, but not for query processor parsing or DLQ parsing. 
2) If there is functionality in this PR that is not ready to be tested yet (like the stream loader config) then ideally you would remove it. I'm not really sure why that's in the PR, you aren't actually parsing it.